### PR TITLE
Add solid-js support

### LINKED ISSUE #5701: Add SolidJS support
### Clear and concise description of the problem

[SolidJS](https://www.solidjs.com/) is the new React, it is getting more users, in the current time SolidJS gets 86K downloads per month.

### Suggested solution

[SolidJS](https://www.solidjs.com/) has a syntax and API very close to React, so the React version of Swiper could be ported.

### Alternative

Just use the Swiper API.

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Would you like to open a PR for this feature?

- [X] I'm willing to open a PR

### PR DESCRIPTION
**Pull request by TiagoCavalcante**
Fixes #5701.

### CODE REVIEW

**Comment by TiagoCavalcante** [comment]
I already have converted the code to SolidJS, but I need help with ESLint, should I disable React plugins on the SolidJS files, or just add `@ts-expect-error`? Also, is there another thing that I need to do to "enable" SolidJS support other than adding the SolidJS bindings?

**Comment by TiagoCavalcante** [comment]
I removed those errors with `// eslint-disable-line` and `/* eslint-disable react/foo-bar */`.

**Comment by TiagoCavalcante** [comment]
Now this is completely working (at least in my tests). I hope this gets merged soon. There are other people in the SolidJS Discord wanting to use it.

**Comment by TiagoCavalcante** [comment]
By now I have published [@tiagotrindade/swiper](https://www.npmjs.com/package/@tiagotrindade/swiper).

**Comment by nolimits4web** [comment]
This is huge, thanks! I am not too familiar with Solid, so it would probably take some time for me to review all of this. Will be also super helpful, if you can  help with Swiper Solid docs in this repo https://github.com/nolimits4web/swiper-website

**Comment by TiagoCavalcante** [comment]
@nolimits4web that will take some time too, but I can help with it.

File: playground/solid/App.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/** @jsxImportSource solid-js */
+/* eslint-disable no-restricted-globals */
+import { Navigation, Pagination, Scrollbar, A11y } from 'swiper';
+import { Swiper, SwiperSlide } from 'swiper/react/swiper-react';
```
Comment (vltansky) [inline playground/solid/App.jsx:4]: shouldn't it import from a solid folder?

File: playground/solid/index.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Swiper React</title>
```
Comment (vltansky) [inline playground/solid/index.html:7]: Change title to Swiper Solid

**Review by vltansky — CHANGES_REQUESTED** [review]

**Comment by vltansky** [comment]
@TiagoCavalcante awesome work!

File: playground/solid/App.jsx (hunk 1 above)
Comment (TiagoCavalcante) [reply playground/solid/App.jsx:4]: Yes, I missed this one.

**Comment by TiagoCavalcante** [comment]
@nolimits4web @vltansky I already rebased this to master but the merge conflicts are still here, I believe this need to be merged from command line.

**Comment by TiagoCavalcante** [comment]
@nolimits4web @vltansky I have started documenting it, but still there is a some (a lot?) of work to do: [swiper-website#259](https://github.com/nolimits4web/swiper-website/pull/259)

**Comment by AndaHendriksen** [comment]
+1

**Comment by nolimits4web** [comment]
hey @TiagoCavalcante can you check Solid playground? It seems doesn't render anything, I guess because `Tag` props are `null` by default but must be `div`s

**Comment by nolimits4web** [comment]
@TiagoCavalcante this is what I see when I run `npm run solid`
ss

**Comment by TiagoCavalcante** [comment]
@nolimits4web I'll fix that, I did a small update and didn't test it again :upside_down_face: 
Wouldn't you mind merging this and I iterate over it in other PRs?
This way maybe someone appears to help with the documentation

**Comment by nolimits4web** [comment]
@TiagoCavalcante this should be fixed before merging, otherwise we can end up having broken Solid JS components in the release

**Comment by TiagoCavalcante** [comment]
@nolimits4web I have fixed it now:
image

**Comment by nolimits4web** [comment]
@TiagoCavalcante thanks, can you check the reactivity now? I added  the following code to the playground:
```jsx
/** @jsxImportSource solid-js */
/* eslint-disable no-restricted-globals */
import { createSignal } from 'solid-js';
import { Navigation, Pagination, Scrollbar, A11y } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/solid/swiper-solid';

const App = () => {
  const [enabled, toggle] = createSignal(true);
  return (
    <main>
      <button onClick={() => toggle(!enabled())}>Toggle</button>
      <p>Enabled: {String(enabled())}</p>
      <Swiper
        modules={[Navigation, Pagination, Scrollbar, A11y]}
        onSwiper={(swiper) => (window.swiper = swiper)}
        slidesPerView={enabled() ? 3 : 1}
        spaceBetween={enabled() ? 50 : 16}
        navigation={enabled()}
        loop
        scrollbar={enabled()}
        pagination={enabled()}
      >
        <SwiperSlide>Slide 1</SwiperSlide>
        <SwiperSlide>Slide 2</SwiperSlide>
        <SwiperSlide>Slide 3</SwiperSlide>
        <SwiperSlide>Slide 4</SwiperSlide>
        <SwiperSlide>Slide 5</SwiperSlide>
      </Swiper>
    </main>
  );
};

export default App;
```
And i assume it should toggle navigation/pagination/scrollbar/slidesPerView/etc. when i click the button, but it doesn't work and no swiper parameters changed on button click

**Comment by TiagoCavalcante** [comment]
@nolimits4web fixed now.

**Review by nolimits4web — APPROVED** [review]

**Review by nolimits4web — APPROVED** [review]

**Pull request closed without merging**